keyboards/J80/j80.c

```diff
@@ -0,0 +1,17 @@
+/* Copyright 2019 JER
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "j80.h"
```

Comment:
Have you traced the lock LED pins? It would be nice to have them here:
```c
bool led_update_kb(led_t led_state) {
    if (led_update_user(led_state)) {
        writePin(???, led_state.caps_lock);
        writePin(???, led_state.num_lock);
        writePin(???, led_state.scroll_lock);
    }
    return true;
}
```
If they are inverted, add a `!` before `led_state` in the `writePin()`s.

Reply:
No, I haven't. And not sure how to do that. Does it use the same pins as the switches would use? On this PCB you can choose between lock leds or switches for print screen, scroll lock and pause.

Reply:
Basically, one end of the LED pins for the switch will have continuity with GND or VCC, and the other end will be connected to a pin on the 32A. The nearby resistor will be in the way on one of the ends.

According to [this image](https://geekhack.org/index.php?action=dlattach;topic=96264.0;attach=206695;image), the num lock and caps lock LEDs are normal backlight LEDs by default, and you can change it by soldering the resistor vertically instead. So what you'd want to do is find which pins the bottom pad of the num/caps resistors, and the left pad of the scroll lock resistor, are connected to.

Similarly, one of the MCU pins will be connected to a small 3-pin chip (a transistor to handle the current that all those LEDs will need). Set that as `BACKLIGHT_PIN` in config.h and we should have backlight support 👍.

Reply:
Thanks for the info. I'll look into this tomorrow.

Reply:
Got the lock LEDs traced. Should I just replace `???` in your code with the pin number?

Not sure where to trace for the `BACKLIGHT_PIN`. Please see attached image and mark where the 3-pin chip is.
![IMG_0143](https://user-images.githubusercontent.com/46304257/69724365-16f2f880-111c-11ea-9a42-0fd0f29cac8a.jpeg)

Reply:
I'm guessing that the 3-pin transistors are the ones on the top to the right. I get continuity from all 3 to different pads on the Lock LEDS. The top and middle are connected to D3 on the MCU and the bottom is connected to D2. D3 is the pin for Scroll Lock, but D2 is not used anywhere else.

Reply:
Given this is a TKL board having two transistors supporting that many backlight LEDs sounds reasonable, but using the same pin for the scroll lock LED doesn't make sense. Do you have a LED to test this out with? You don't have to solder it in, just hold the leads against the pads and try toggling the lock keys and changing the backlight level (it's currently set to 1 in config.h, you'll need to increase that). The longer lead goes in the square pad marked +.

Reply:
Sorry, don't have any LED's on hand. But traced the Lock LED's again and turns out I messed up the pin on Scroll Lock, it's actually D6.

The top two transistors are connected to D3 and the bottom to D2. Is D3 the backlight pin?

Also is this code correct for j80.c
`bool led_update_kb(led_t led_state) {
    if (led_update_user(led_state)) {
        writePin(D1, led_state.caps_lock);
        writePin(D0, led_state.num_lock);
        writePin(D6, led_state.scroll_lock);
    }
    return true;
}`

Reply:
Okay, that makes way more sense, it's starting to match the usual PS2AVRGB/Bootmapper LED pins per [this schematic](https://github.com/showjean/ps2avrU/blob/master/firmware/ps2avrGB/Circuit/ps2avrGB/20140829_140848_001.png) (this seems to be common to at least all the Winkeyless boards, as the firmware I believe is a one-size-fits-all type deal).
They would then be:
Num Lock D0
Caps Lock D1
Backlight D4
Scroll Lock D6

The two other transistors have connections to the USB data lines (D2/D3) and D5, and have something to do with detecting or switching between USB and PS/2. Since QMK only does USB AFAIK, we don't have to worry about that.

Reply:
Nice. Commited changes for backlight and lock LED support. Too bad I don't have any LED's to test with.

Reply:
Got some LED's with another PCB. Tested backlight and lock lights, all working as expected.